src/coreclr/jit/lclvars.cpp

```diff
@@ -6739,7 +6739,7 @@ int Compiler::lvaToCallerSPRelativeOffset(int offset, bool isFpBased, bool forRo
         offset += codeGen->genCallerSPtoInitialSPdelta();
     }
 
-#if defined(TARGET_AMD64) || defined(TARGET_ARM64) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
+#ifdef TARGET_64BIT
```

Comment:
How does `FEATURE_ON_STACK_REPLACEMENT` break the build here? (The build log is no longer available.)

Reply:
It was failing for x86_x64 variant of altjit, because `FEATURE_ON_STACK_REPLACEMENT` was true but TARGET_AMD64 and ARM64 were false:

```
  [ 82%] Building CXX object tools/superpmi/superpmi/CMakeFiles/superpmi.dir/__/superpmi-shared/simpletimer.cpp.o
  /__w/1/s/src/coreclr/jit/lclvars.cpp:6766:19: error: use of undeclared identifier 'adjustment'
   6766 |         offset -= adjustment;
        |                   ^
  [ 82%] Building CXX object jit/CMakeFiles/clrjit_win_x86_x64.dir/likelyclass.cpp.o
```
(I have devops page opened in a tab 🙂)

Reply:
I think this means that `FEATURE_ON_STACK_REPLACEMENT` is not defined correctly for cross-targeting JITs.

Reply:
Looks like all features in clrdefinitions.cmake are defined based on the target, so cross-targeting altjits are not respecting the features which are tied to the target (instead of host). That requires explicit `TARGET_*` checks, in addition to the feature flags.

@jakobbotsch, @BruceForstall, is it by design?

Reply:
I guess we don't have many JIT feature ifdefs that we try to define via cmake. Most JIT feature ifdefs are defined in the `target*.h` files. For example:

https://github.com/dotnet/runtime/blob/cc3700953542b96052f73fc7ee259994692575cf/src/coreclr/jit/targetarm64.h#L19-L33